webhooks/provisioning_webhook.go

```diff
@@ -0,0 +1,86 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhooks
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	metal3iov1alpha1 "github.com/openshift/cluster-baremetal-operator/apis/metal3.io/v1alpha1"
+	"github.com/openshift/generic-admission-server/pkg/apiserver"
+)
+
+type ProvisioningValidatingWebHook struct {
+}
+
+// +kubebuilder:rbac:groups=flowcontrol.apiserver.k8s.io,resources=prioritylevelconfigurations;flowschemas,verbs=get;list;watch
+// +kubebuilder:rbac:groups=admission.metal3.io,resources=provisionings,verbs=create
+
+var _ apiserver.AdmissionHook = &ProvisioningValidatingWebHook{}
+
+func (a *ProvisioningValidatingWebHook) ValidatingResource() (plural schema.GroupVersionResource, singular string) {
+	return schema.GroupVersionResource{
+		Group:    "admission.metal3.io",
```

Comment:
the amount of time i took to figure out i had this ^ wrong is mind boggling

Reply:
Perhaps the team shouldn't adopt library-go until there is better documentation for using it?

Reply:
It was in the example (so my bad), but I am happy to add docs..
Also this is not really library-go's fault as it is using https://github.com/openshift/generic-admission-server for the webhook

Reply:
https://github.com/openshift/library-go/pull/1066